pkg/agent/baker.go

```diff
@@ -1249,6 +1249,33 @@ root = "{{GetDataDir}}"{{- end}}
     type = "snapshot"
     address = "/run/overlaybd-snapshotter/overlaybd.sock"
 {{- end}}
+{{- if IsKata }}
+[plugins."io.containerd.grpc.v1.cri".containerd.runtimes.kata]
+  runtime_type = "io.containerd.kata.v2"
+[plugins."io.containerd.grpc.v1.cri".containerd.runtimes.katacli]
+  runtime_type = "io.containerd.runc.v1"
+[plugins."io.containerd.grpc.v1.cri".containerd.runtimes.katacli.options]
+  NoPivotRoot = false
+  NoNewKeyring = false
+  ShimCgroup = ""
+  IoUid = 0
+  IoGid = 0
+  BinaryName = "/usr/bin/kata-runtime"
+  Root = ""
+  CriuPath = ""
+  SystemdCgroup = false
+[proxy_plugins]
+  [proxy_plugins.tardev]
+    type = "snapshot"
+    address = "/run/containerd/tardev-snapshotter.sock"
+[plugins."io.containerd.grpc.v1.cri".containerd.runtimes.kata-cc]
```

Comment:
I only did a high level review. This is what I am looking for. Once this handler is in, I will change AKS code to use this new handler.

The IsKata condition is correct in that Kata-cc still use the same distro/image as Kata.